src/Workspaces/Core/Portable/SourceGeneration/IRemoteSourceGenerationService.cs

```diff
@@ -57,6 +57,13 @@ ValueTask<bool> HasGeneratorsAsync(
     /// </summary>
     ValueTask<ImmutableArray<SourceGeneratorIdentity>> GetSourceGeneratorIdentitiesAsync(
         Checksum solutionChecksum, ProjectId projectId, string analyzerReferenceFullPath, CancellationToken cancellationToken);
+
+    /// <summary>
+    /// Returns whether or not the the <see cref="AnalyzerReference"/> with <see cref="AnalyzerFileReference.FullPath"/>
+    /// equal to <paramref name="analyzerReferenceFullPath"/> has any analyzers or source generators.
+    /// </summary>
+    ValueTask<bool> HasAnalyzersOrSourceGeneratorsAsync(
+        Checksum solutionChecksum, ProjectId projectId, string analyzerReferenceFullPath, CancellationToken cancellationToken);
```

Comment:
this was the core goal of hte PR.  The VS SolutionExplroer side should defer to it to answer this question, instead of loading analyzers/generators in proc (which doesn't work with reloading with .Net Framework).